stdlib/public/core/UnicodeScalarProperties.swift

```diff
@@ -694,26 +694,25 @@ extension Unicode.Scalar.Properties {
   internal func _applyMapping(_ u_strTo: _U_StrToX) -> String {
     // TODO(String performance): Stack buffer first and then detect real count
     let count = 64
-    var array = Array<UInt16>(repeating: 0, count: count)
-    let len: Int = array.withUnsafeMutableBufferPointer { bufPtr in
-      return _scalar.withUTF16CodeUnits { utf16 in
-        var err = __swift_stdlib_U_ZERO_ERROR
+    let array = Array<UInt16>(unsafeUninitializedCapacity: count) {
+      buffer, initializedCount in
+      initializedCount = _scalar.withUTF16CodeUnits { utf16 in
+        var error = __swift_stdlib_U_ZERO_ERROR
         let correctSize = u_strTo(
-          bufPtr.baseAddress._unsafelyUnwrappedUnchecked,
-          Int32(bufPtr.count),
+          buffer.baseAddress._unsafelyUnwrappedUnchecked,
+          Int32(buffer.count),
           utf16.baseAddress._unsafelyUnwrappedUnchecked,
           Int32(utf16.count),
           "",
-          &err)
-        guard err.isSuccess else {
+          &error)
+        guard error.isSuccess else {
           fatalError("Unexpected error case-converting Unicode scalar.")
         }
-        // TODO: _internalInvariant(count == correctSize, "inconsistent ICU behavior")
+        _internalInvariant(count == correctSize, "inconsistent ICU behavior")
```

Comment:
I don't know the history of this TODO, so I'm not certain it's safe to uncomment this invariant. @milseman?